anylabeling/services/auto_labeling/yolov5_seg.py

```diff
@@ -0,0 +1,244 @@
+                scores[i]
+            )  # get score and classId
+
+            if (
+                score >= self.config["score_threshold"]
+            ):  # filtering by score_tresh
+                if self.output_mode == "polygon":
+                    # Transform coordinates
+                    left = int(round(box[0] * seg_width / image_width))
+                    top = int(round(box[1] * seg_height / image_height))
+                    width = int(round(box[2] * seg_width / image_width))
+                    height = int(round(box[3] * seg_height / image_height))
+
+                    # Process protos
+                    protos = outputs[1][
+                        0, :, top : top + height, left : left + width
+                    ].reshape(seg_channels, -1)
+                    protos = np.expand_dims(masks[i], 0) @ protos  # matmul
+                    protos = 1 / (1 + np.exp(-protos))  # sigmoid
+                    protos = protos.reshape(height, width)  # reshape
+                    mask = cv2.resize(protos, (box[2], box[3]))  # resize mask
+                    mask = (
+                        mask >= self.config["mask_threshold"]
+                    )  # filtering mask by tresh
+
+                    # Add mask to overlay layer
+                    black = np.zeros((image_height, image_width, 3))
+                    to_mask = black[
+                        box[1] : box[1] + box[3], box[0] : box[0] + box[2]
+                    ]  # get box roi
+                    to_mask[mask] = [255, 255, 255]  # apply mask
+
+                    gray = cv2.cvtColor(
+                        black.astype(np.uint8), cv2.COLOR_BGR2GRAY
+                    )
+                    contours, hierarchy = cv2.findContours(
+                        gray.astype(np.uint8),
+                        cv2.RETR_EXTERNAL,
+                        cv2.CHAIN_APPROX_NONE,
+                    )
+
+                    # Refine contours (taken from segment_anything.py)
+                    approx_contours = []
+                    for contour in contours:
+                        # Approximate contour
+                        epsilon = 0.001 * cv2.arcLength(contour, True)
+                        approx = cv2.approxPolyDP(contour, epsilon, True)
+                        approx_contours.append(approx)
+
+                    # Remove small contours (area < 20% of average area)
+                    if len(approx_contours) > 1:
+                        areas = [
+                            cv2.contourArea(contour)
+                            for contour in approx_contours
+                        ]
+                        avg_area = np.mean(areas)
+
+                        filtered_approx_contours = [
+                            contour
+                            for contour, area in zip(approx_contours, areas)
+                            if area > avg_area * 0.2
+                        ]
+                        contours = filtered_approx_contours
+
```

Comment:
Please add the first point again as the last element to close the contour.
Like the line 271 of this file: <https://github.com/vietanhdev/anylabeling/blob/master/anylabeling/services/auto_labeling/segment_anything.py>
```
points.append(points[0])
```
The contours will not be closed if you don't do it.

![image](https://user-images.githubusercontent.com/18329471/234012461-9be02cb9-91a7-4456-8a6d-bb9d6ba8c8e2.png)

Reply:
https://user-images.githubusercontent.com/18329471/234012461-9be02cb9-91a7-4456-8a6d-bb9d6ba8c8e2.png